packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -147,12 +177,34 @@ export default function createCssVarsProvider(options) {
         shouldSkipGeneratingVar,
       });
       theme.vars = deepmerge(theme.vars, vars);
+
       if (key === calculatedColorScheme) {
+        const restParsedScheme = { ...parsedScheme };
```

Comment:
```suggestion
```
The `parsedScheme` is already a new object, creating a new one makes it harder to read.

Reply:
Refactored, based on https://github.com/mui/material-ui/pull/34650#discussion_r1022316372